warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/EventMapper.java

```diff
@@ -792,21 +793,36 @@ public Multimap<String,NormalizedContentInterface> getFields(RawRecordContainer
         if (handler.getHelper(value.getDataType()) instanceof VirtualIngest) {
             VirtualIngest vHelper = (VirtualIngest) handler.getHelper(value.getDataType());
             Multimap<String,NormalizedContentInterface> virtualFields = vHelper.getVirtualFields(newFields);
-            for (Entry<String,NormalizedContentInterface> v : virtualFields.entries())
+            for (Entry<String,NormalizedContentInterface> v : virtualFields.entries()) {
                 newFields.put(v.getKey(), v.getValue());
+            }
         }
+
         // Also get the composite fields, if applicable
         if (handler.getHelper(value.getDataType()) instanceof CompositeIngest) {
             CompositeIngest vHelper = (CompositeIngest) handler.getHelper(value.getDataType());
             Multimap<String,NormalizedContentInterface> compositeFields = vHelper.getCompositeFields(newFields);
             for (String fieldName : compositeFields.keySet()) {
                 // if this is an overloaded composite field, we are replacing the existing field data
-                if (vHelper.isOverloadedCompositeField(fieldName))
+                if (vHelper.isOverloadedCompositeField(fieldName)) {
                     newFields.removeAll(fieldName);
+                }
                 newFields.putAll(fieldName, compositeFields.get(fieldName));
             }
         }
 
+        // Also get whindex fields, if applicable
+        if (handler.getHelper(value.getDataType()) instanceof WhindexIngest) {
+            WhindexIngest vHelper = (WhindexIngest) handler.getHelper(value.getDataType());
+            Multimap<String,NormalizedContentInterface> whindexFields = vHelper.getWhindexFields(newFields);
+            for (String fieldName : whindexFields.keySet()) {
+                if (vHelper.isOverloadedWhindexField(fieldName)) {
```

Comment:
I think we need to rework how overloaded is implemented here.  We should only be droppping source fields IFF they were turned into whindex/destination fields.